### PR TITLE
chore: Making feature-deployment with mocks an explicit decision

### PR DESCRIPTION
**Pull request by pshomov**
Making feature deployments with mocks an explicit option. Massaging the local environment generation.
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: infra/src/dsl/types/output-types.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ export type HelmValueFile = {
 }
 export type LocalrunValueFile = {
   services: Services<LocalrunService>
-  mocks: Services<LocalrunService>
+  mocks: string
```
Comment (brynjarorng) [inline infra/src/dsl/types/output-types.ts:184]: Is this so that new services can be added without changing the interface?

**Review by brynjarorng — APPROVED** [review]
LGTM(Devops)

File: apps/system-e2e/global.d.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // global.d.ts
+export {};
+
```
Comment (kristofer-lp8f) [inline apps/system-e2e/global.d.ts:3]: What's this for?

File: infra/src/feature-env.ts [hunk 3]
```diff
@@ -117,7 +117,7 @@ yargs(process.argv.slice(2))
         env,
       )
       await writeToOutput(
-        await renderHelmValueFileContent(env, habitat, featureYaml, 'no-mocks'),
+        await renderHelmValueFileContent(env, habitat, featureYaml, argv.withMocks ?? false ? 'with-mocks' : 'no-mocks'),
```
Comment (kristofer-lp8f) [inline infra/src/feature-env.ts:120]:
```suggestion
        await renderHelmValueFileContent(env, habitat, featureYaml, argv.withMocks ? 'with-mocks' : 'no-mocks'),
```

`undefined` is also falsey

**Review by kristofer-lp8f — APPROVED** [review]

File: infra/src/dsl/types/output-types.ts (hunk 1 above)
Comment (pshomov) [reply infra/src/dsl/types/output-types.ts:184]: No, I am just adjusting the output for the mocks section - it is a command string to be copy/pasted into the terminal.

File: apps/system-e2e/global.d.ts (hunk 2 above)
Comment (pshomov) [reply apps/system-e2e/global.d.ts:3]: So that TS stops complaining about `.toHaveCountGreaterThan()`

File: infra/src/feature-env.ts (hunk 3 above)
Comment (pshomov) [reply infra/src/feature-env.ts:120]: True, still I prefer to make the default explicit.

**Pull request merged**